pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -236,7 +236,9 @@ func (m *Matrix) countGeneratedCombinationsFromParams() int {
 	}
 	count := 1
 	for _, param := range m.Params {
-		count *= len(param.Value.ArrayVal)
+		if len(param.Value.ArrayVal) > 0 {
+			count *= len(param.Value.ArrayVal)
```

Comment:
nit: can we just go with `count = count * len(…)` ?

Reply:
```shell
Running [/home/runner/golangci-lint-1.62.2-linux-amd64/golangci-lint run  --timeout=10m] in [/home/runner/work/pipeline/pipeline] ...

Error: pkg/apis/pipeline/v1/matrix_types.go:240:4: assignOp: replace `count = count * len(param.Value.ArrayVal)` with `count *= len(param.Value.ArrayVal)` (gocritic)
```

Reply:
Intesresting 🙃